doc/integrations/webhooks/Incoming-Webhooks.md

```diff
@@ -0,0 +1,74 @@
+| Component  | Tests Run   | Tests Failed                                   |
+|:-----------|:------------|:-----------------------------------------------|
+| Server     | 948         | :white_check_mark: 0                           |
+| Web Client | 123         | :warning: [2 (see details)](http://linktologs) |
+| iOS Client | 78          | :warning: [3 (see details)](http://linktologs) |
+***
+"}
+```
+Which would render in a Mattermost message as follows:
+
+*** 
+##### Build Break - Project X - December 12, 2015 - 15:32 GMT +0
+| Component  | Tests Run   | Tests Failed |
+|:------------ |:---------------|:-----|
+| Server | 948       |  :white_check_mark: 0 |
+| Web Client | 123              |   :warning: [2 (see details)](http://linktologs) |
+| iOS Client | 78        |    :warning: [3 (see details)](http://linktologs) |
+*** 
+
+### Creating Integrations using Incoming Webhooks
+You can create a webhook integration to post into Mattermost channels and private groups using these steps: 
+
+1. Enable incoming webhooks from **System Console -> Service Settings**  
+
+ 1. Optionally configure the **Enable Overriding of Usernames from Webhooks** option to allow external applications to post messages under any name. If not enabled, the username of the creator of the webhook URL is used to post messages.
+ 2. Optionally configure the **Enable Overriding of Icon from Webhooks** option to allow external applciations to change the icon of the account posting messages. If not enabled, the icon of the creator of the webhook URL is used to post messages. 
+2. Create a Mattermost Incoming Webhook URL
+ 1. Login to your Mattermost team site and go to **Account Settings -> Integrations**
+ 2. Next to **Incoming Webhooks** click **Edit**
+ 3. Select the channel or private group to receive webhook payloads, then click **Add** to create the webhook
+
+3. Write a function to call the URL with a properly formatted JSON payload
+ 1. To make sure everything works, try a curl command to send a JSON string as the `payload` parameter in a HTTP POST request.
+     1. Example: 
+     ```
+curl -i -X POST -d 'payload={"text": "Hello, this is some text."}' http://yourmattermost.com/hooks/xxx-generatedkey-xxx
+```
+ 2. Incorporate the working webhook into your application to send real time updates to Mattermost channels and private groups.
+
+Additional Notes:
+
+1. For the JSON payload, if `Content-Type` is specified as `application/json` in the headers of the HTTP request then the body of the request can be direct JSON. ```{"text": "Hello, this is some text."}```  
+
+2. You can override the channel specified in the webhook definition by specifying a `channel` parameter in your payload. For example, you might have a single webhook created for _Town Square_, but you can use ```payload={"channel": "off-topic", "text": "Hello, this is some text."}``` to send a message to the _Off-Topic_ channel using the same webhook URL.  
```

Comment:
Some additional notes from documentation I was planning on adding this morning:

1. In addition, with **Enable Overriding of Usernames from Webhooks** turned on,  you can also override the username the message posts as by providing a `username` parameter in your JSON payload. For example, you might want your message looking like it came from a robot so you can use ```payload={"username": "robot", "text": "Hello, this is some text."}``` to change the username of the post to robot. Note, to combat any malicious users from trying to use this to perform [phishing attacks](https://en.wikipedia.org/wiki/Phishing) a `BOT` indicator appears next to posts coming from incoming webhooks.

2. With **Enable Overriding of Icon from Webhooks** turned on, you can similarly change the icon the message posts with by providing a link to an image in the `icon_url` parameter of your payload. For example, ```payload={"icon_url": "http://somewebsite.com/somecoolimage.jpg", "text": "Hello, this is some text."}``` will post using whatever image is located at `http://somewebsite.com/somecoolimage.jpg` as the icon for the post.